subiquity/server/controllers/install.py

```diff
@@ -210,45 +218,42 @@ async def curtin_install(self, *, context, source):
 
         resume_data_file = Path(tempfile.mkdtemp()) / "resume-data.json"
 
-        await self.run_curtin_install_step(step=CurtinInstallStep(
-                name="initial",
-                stages=[],
-                config_file=config_dir / "subiquity-initial.conf",
+        def make_curtin_step(name, stages, acquire_config):
+            return CurtinInstallStep(
+                controller=self,
+                name=name,
+                stages=stages,
+                config_file=config_dir / f"subiquity-{name}.conf",
                 log_file=install_log_file,
                 error_file=error_file,
-                acquire_config=self.acquire_initial_config,
-            ), resume_data_file=resume_data_file,
-            context=context, source=source)
+                resume_data_file=resume_data_file,
+                source=source,
+                acquire_config=acquire_config,
+            )
+
+        await make_curtin_step(
+            "initial",
+            stages=[],
+            acquire_config=self.acquire_initial_config).run(context=context)
 
         generic_steps = [
```

Comment:
if we're storing callables here, I think we don't need to treat the "initial" step differently - and we can drop the _generic_ prefix from this variable.

Reply:
Well one of the very next things to do is to start passing different configs to different invocations. Possibly a caller here isn't the best way to do it but it's sufficient I guess?

Reply:
Yes, a caller/callable is fine IMO.

What I meant is that we could add have all the steps (including the initial step in the same sequence):

```diff
-         await make_curtin_step(
-           "initial",
-            stages=[],
-            acquire_config=self.acquire_initial_config).run(context=context)
-
-        generic_steps = [
+       steps: List[Callable[Context], None] = [
+              make_curtin_step(name="initial", stages=[], acquire_config=self.acquire_initial_config).run
               make_curtin_step(name="partitioning", stages=["partitioning"], acquire_config=self.acquire_generic_config).run
               make_curtin_step(name="extract", stages=["extract"], acquire_config=self.acquire_generic_config).run
```

Reply:
oh right, yes that makes sense

Reply:
thanks!